### PR TITLE
Restructure the home page into a user guide

### PR DESCRIPTION
**Pull request by llucax**
This commit split the `docs/index.md` file, that grew quite significantly, into a more organized user guide top-level section.

The content is the same, it is just reorganized.

To make the new user guide sub-sections more visible we now configure mkdocs to automatically expand the navigation bar.

### CODE REVIEW

**Comment by llucax** [comment]
Made a draft because it is based on:

* #192

**Comment by llucax** [comment]
#192 merged, updating, so ready for review.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 ## New Features
 
 - New GitHub rulesets are provided with the recommended configuration to protect branches and tags.
+- The documentation is restructure in a more organized, easier to navigate, user guide.
```
Comment (daniel-zullo-frequenz) [inline RELEASE_NOTES.md]:
```suggestion
- The documentation is restructured into a more organized, easier-to-navigate user guide.
```

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I only have an optional cosmetic comment. LGTM

File: RELEASE_NOTES.md (hunk 1 above)
Comment (llucax) [reply RELEASE_NOTES.md]: Is this easier-to-read style really used? Shouldn't the last comma be kept?

Comment (daniel-zullo-frequenz) [reply RELEASE_NOTES.md]: Yes, I think the hyphened style is commonly used to combine 2 or 3 words when they work together as an adjective like in easier-to-read, easier-to-navigate, user-friendly, etc.
I don't think the comma is needed but I might be wrong about it.

**Comment by llucax** [comment]
Accepted the suggestion, will merge.

**Pull request merged**